internal/weldr/api.go

```diff
@@ -1590,6 +1590,29 @@ func setPkgEVRA(dependencies []rpmmd.PackageSpec, packages []blueprint.Package)
 	return nil
 }
 
+// expandBlueprintGlobs will expand any package name globs into the matches from the dependencies.
+func expandBlueprintGlobs(dependencies []rpmmd.PackageSpec, packages []blueprint.Package) ([]blueprint.Package, error) {
+	var newPackages []blueprint.Package
+	for _, pkg := range packages {
+		if !strings.ContainsAny(pkg.Name, "*?") {
+			// No glob characters, pass it through
+			newPackages = append(newPackages, pkg)
+		} else {
+			// Add all the packages matching the glob
+			g, err := glob.Compile(pkg.Name)
+			if err != nil {
+				return nil, err
+			}
+			for _, d := range dependencies {
+				if g.Match(d.Name) {
+					newPackages = append(newPackages, blueprint.Package{Name: d.Name, Version: pkg.Version})
+				}
+			}
+		}
+	}
+	return newPackages, nil
```

Comment:
Shall we deduplicate the list before returning? I think that more items from `packages` can match one package from `dependencies`?